src/openforms/forms/tests/e2e_tests/test_form_designer.py

```diff
@@ -216,3 +217,201 @@ def setUpTestData():
                 "worden afgesloten met een streepje of punt."
             )
             await expect(parent).to_contain_text(error_message)
+
+
+class FormDesignerRegressionTests(E2ETestCase):
+    async def test_user_defined_variable_boolean_initial_value_false(self):
+        """
+        Regression test for https://github.com/open-formulieren/open-forms/issues/2636
+        """
+
+        @sync_to_async
+        def setUpTestData():
+            # set up a form
+            form = FormFactory.create(
+                name="Playwright test",
+                name_nl="Playwright test",
+                formstep__form_definition__name_nl="Playwright test",
+            )
+            return form
+
+        await create_superuser()
+        form = await setUpTestData()
+        admin_url = str(
+            furl(self.live_server_url)
+            / reverse("admin:forms_form_change", args=(form.pk,))
+        )
+
+        async with browser_page() as page:
+            await self._admin_login(page)
+            await page.goto(str(admin_url))
+            await page.get_by_role("tab", name="Variables").click()
+            await page.get_by_role("tab", name="User defined").click()
+            with phase("Add variable"):
+                await page.get_by_text("Add variable").click()
+                await page.locator("#id_name").fill("Foo")
+                await page.locator("#id_name").blur()
+                await page.locator("#id_dataType").select_option(label="Boolean")
+                await page.locator("[name='initialValue']").select_option(label="No")
+
+                # Verify that the select updated to the selected value
+                await expect(page.locator("[name='initialValue']")).to_have_value(
+                    "false"
+                )
+
+            with phase("Save variable and check state"):
+                await page.get_by_text("Save and continue editing").click()
+                await page.get_by_role("tab", name="Variabelen").click()
+                await page.get_by_role("tab", name="Gebruikersvariabelen").click()
+
+                # Verify that the select still holds the correct value
+                await expect(page.locator("[name='initialValue']")).to_have_value(
+                    "false"
+                )
```

Comment:
can we also assert that the label of the dropdown is "No"?

Reply:
I tried doing this (with `.to_have_text`), but that seems to just concatenate the text of all the possible options together (e.g. `------YesNo`)